test/common.hpp

```diff
@@ -40,8 +43,27 @@ namespace menoh_impl {
     auto assert_near_list(List1 const& list1, List2 const& list2, float eps) {
         using std::begin;
         using std::end;
-        assert_near_list(begin(list1), end(list1), begin(list2), end(list2), eps);
+        assert_near_list(begin(list1), end(list1), begin(list2), end(list2),
+                         eps);
+    }
+
+    inline bool is_near_array(array const& lhs, array const& rhs,
+                              float eps = 1e-4) {
+        assert(lhs.dtype() == dtype_t::float_);
+        assert(rhs.dtype() == dtype_t::float_);
+        if(total_size(lhs) != total_size(rhs)) {
+            return false;
+        }
+        auto size = total_size(lhs);
+        for(auto i = 0; i < size; ++i) {
+            float l = *(static_cast<float*>(lhs.data()) + i);
+            float r = *(static_cast<float*>(rhs.data()) + i);
+            if(!(l - eps < r && r < l + eps)) {
+                return false;
+            }
+        }
+        return true;
```

Comment:
Is this necessary for the objective of this PR?
If not, how about removing this from this PR and applying the change separately?

Reply:
`is_near_array` is used for testing [here](https://github.com/pfnet-research/menoh/pull/88/files#diff-a07ff2d45ccdec4e633eb369f99677bbR91)

Reply:
Oop! I overlooked that.
Thanks!

Reply:
I will add some comments into testing code.